src/file.rs

```diff
@@ -13,246 +13,225 @@ use FileQueryInfoFlags;
 use ffi;
 use glib::object::IsA;
 use glib::translate::*;
+use glib::Priority;
 use glib_ffi;
 use gobject_ffi;
-use std::mem::transmute;
 use std::ptr;
 use File;
 
 pub trait FileExtManual {
-    fn append_to_async<'a, P: Into<Option<&'a Cancellable>>, Q: Fn(Result<FileOutputStream, Error>) + Send + Sync + 'static>(&self, flags: FileCreateFlags, io_priority: i32, cancellable: P, callback: Q);
+    fn append_to_async<'a, P: Into<Option<&'a Cancellable>>, Q: FnOnce(Result<FileOutputStream, Error>) + Send + 'static>(&self, flags: FileCreateFlags, io_priority: Priority, cancellable: P, callback: Q);
```

Comment:
Why are *all* these manual? Only because of the priority? I'd rather add support to gir for this instead, this is a huge amount of manual code otherwise :)

Reply:
I think that it part of https://github.com/gtk-rs/gir/issues/504

Reply:
Not really, but it's a missing feature in gir indeed. It's a normal GIO-style async function not handled by gir yet